src/main/java/kitchenpos/order/application/OrderValidator.java

```diff
@@ -0,0 +1,60 @@
+package kitchenpos.order.application;
+
+import kitchenpos.menu.application.MenuService;
+import kitchenpos.menu.domain.Menu;
+import kitchenpos.menu.domain.MenuProduct;
+import kitchenpos.order.domain.Order;
+import kitchenpos.order.domain.OrderLineItem;
+import kitchenpos.table.domain.OrderTable;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.*;
+import static java.util.stream.Collectors.toMap;
+
+@Component
+public class OrderValidator {
+
+    private MenuService menuService;
+
+    public OrderValidator(final MenuService menuService) {
+        this.menuService = menuService;
+    }
+
+    public void validate(Order order) {
+        validate(order, getMenus(order));
+    }
+
+    private void validate(Order order, Map<Long, Menu> menus) {
+
+        List<OrderLineItem> orderLineItems = order.getOrderLineItems();
+
+        for(OrderLineItem item : orderLineItems) {
+            validateOrderLineItem(item, menus.get(item.getMenuId()));
+        }
+    }
+
+    private void validateOrderLineItem(OrderLineItem item, Menu menu) {
+        final int correctStandard = 0;
+        if(item.getMenuId().equals(menu.getId()) && !item.getMenuName().equals(menu.getName())) {
```

Comment:
```suggestion
        if (item.getMenuId().equals(menu.getId()) && !item.getMenuName().equals(menu.getName())) {
```

`OrderLineItem`한테 물어보시죠 😄 
디미터 법칙에 대해서 한번 정리하고 가시면 좋을 것 같습니다 😄 

도움 되실 만한 내용 공유 드립니다 🙇 
> https://dundung.tistory.com/203

Reply:
객체 지향 생활 체조의 한줄에 . 하나만 찍는다라는 원칙이 Stream api를 사용할 시 말하는 줄 착각했었네요ㅎㅎ